frontend/packages/knative-plugin/src/utils/knative-topology-utils.ts

```diff
@@ -106,31 +105,43 @@ const createKnativeDeploymentItems = (
 /**
  * Form Node data for revisions/event/service sources
  */
-export const getKnativeTopologyNodeItem = (
+export const getKnativeTopologyNodeItems = (
   resource: K8sResourceKind,
   type: string,
   resources?: TopologyDataResources,
-): Node => {
+): Node[] => {
+  const nodes = [];
   const uid = _.get(resource, ['metadata', 'uid']);
   const name = _.get(resource, ['metadata', 'name']);
   const label = _.get(resource, ['metadata', 'labels', 'app.openshift.io/instance']);
   const children = [];
-  if (type === nodeType.KnService && resources && resources.configurations) {
+  if (type === NodeType.KnService && resources && resources.configurations) {
     const configurations = getOwnedResources(resource, resources.configurations.data);
     const configUidData = _.get(configurations[0], ['metadata', 'uid']);
     const ChildData = _.filter(resources.revisions.data, {
-      metadata: { ownerReferences: [{ uid: configUidData }] },
+      metadata: {
+        ownerReferences: [{ uid: configUidData }],
+      },
     });
     _.forEach(ChildData, (c) => {
-      children.push(_.get(c, ['metadata', 'uid']));
+      const nameRev = _.get(c, ['metadata', 'name']);
+      const labelRev = _.get(c, ['metadata', 'labels', 'app.openshift.io/instance']);
+      const uidRev = _.get(c, ['metadata', 'uid']);
+      children.push(uidRev);
+      nodes.push({
+        id: uidRev,
+        type: NodeType.Revision,
+        name: labelRev || nameRev,
```

Comment:
you can use `getTopologyNodeItem` utility from topology-utils and pass type and children as optional param.